presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java

```diff
@@ -135,6 +137,16 @@ public ListSelectiveStreamReader(
         this.outputType = (ArrayType) outputType.orElse(null);
         this.level = subfieldLevel;
 
+        maxElementLength = subfields.stream()
+                .map(Subfield::getPath)
```

Comment:
can we combine all the map into a single one

Reply:
we usually do the opposite

Reply:
since there are 2 maps back to back thought we can combine them, since we iterate every time for a map, but i guess the performance impact will be negligible. so its ok.
```
  int maxElementLength = subfields.stream()
                .map(subfield -> subfield.getPath().get(0))
                .filter(pathElement -> !(pathElement instanceof Subfield.AllSubscripts))
                .map(subfield -> ((Subfield.LongSubscript)subfield).getIndex())
                .mapToInt(Long::intValue)
                .max()
                .orElse(ELEMENT_LENGTH_UNBOUNDED);
```